packages/react/src/analytics/integrations/AnalyticsApi/AnalyticsApi.ts

```diff
@@ -0,0 +1,107 @@
+import {
+  type AnalyticsApiIntegrationOptions,
+  type AnalyticsApiTrackData,
+} from './types/types.js';
+import { client } from '@farfetch/blackout-client';
+import {
+  type EventData,
+  EventTypes,
+  integrations,
+  type LoadIntegrationEventData,
+  PageTypes,
+  type StrippedDownAnalytics,
+  type TrackTypesValues,
+  utils,
+} from '@farfetch/blackout-analytics';
+import { isArray } from 'lodash';
+
+export default class AnalyticsAPI extends integrations.Integration<AnalyticsApiIntegrationOptions> {
+  static override [utils.CONSENT_CATEGORIES_PROPERTY] =
+    utils.DefaultConsentKeys.MARKETING;
+  private debugMode: boolean | undefined;
+
+  /**
+   * Creates an instance of Analytics Api integration.
+   *
+   * @param options               - User configured options.
+   * @param loadData              - Analytics's load event data.
+   * @param strippedDownAnalytics - Analytics stripped down instance.
+   */
+  constructor(
+    options: AnalyticsApiIntegrationOptions,
+    loadData: LoadIntegrationEventData,
+    strippedDownAnalytics: StrippedDownAnalytics,
+  ) {
+    super(options, loadData, strippedDownAnalytics);
+
+    if (
+      this.options.whitelistedEvents &&
+      !isArray(this.options.whitelistedEvents)
+    ) {
+      throw new Error(
+        '[Analytics API] - The value `options.whitelistedEvents` from Analytics Api integration options must be an array.',
+      );
+    }
+
+    if (
+      this.options.blacklistedEvents &&
+      !isArray(this.options.blacklistedEvents)
+    ) {
+      throw new Error(
+        '[Analytics API] - The value `options.blacklistedEvents` from Analytics Api integration options must be an array.',
+      );
+    }
+
+    this.debugMode =
+      utils.getCookie('analyticsAPIDebug')?.toLowerCase() === 'true';
+  }
+
+  /**
+   * Track method that will call Analytics API with the necessary data per event.
+   * If there's more than one pageType per event, it will call Analytics API with each page type at a time.
+   *
+   * @param data - Track event data.
+   *
+   * @returns Promise that will resolve when the method finishes.
+   */
+  override async track(data: EventData<TrackTypesValues>) {
+    const whitelisted = this.options.whitelistedEvents || [
+      ...Object.values(PageTypes),
+      ...Object.values(EventTypes),
+    ];
+    const filteredEvents = whitelisted?.filter(
+      event => !this.options.blacklistedEvents?.includes(event),
+    ) as Array<string>;
+
+    if (filteredEvents.includes(data.event)) {
+      const payload = {
+        data,
+        options: this.getOptions(),
+      };
+
+      if (this.debugMode) {
+        utils.logger.info('ANALYTICS API track:', payload);
```

Comment:
Use the same format you use for other messages:

```suggestion
        utils.logger.info('[Analytics API] - track:', payload);
```